apps/daimo-mobile/src/logic/passkey.ts

```diff
@@ -0,0 +1,194 @@
+import { Buffer } from "@craftzdog/react-native-buffer";
+import { contractFriendlyKeyToDER, parseAndNormalizeSig } from "@daimo/common";
+import { DaimoChain, daimoAccountABI } from "@daimo/contract";
+import * as ExpoPasskeys from "@daimo/expo-passkeys";
+import { SigningCallback } from "@daimo/userop";
+import { base64 } from "@scure/base";
+import {
+  Hex,
+  bytesToHex,
+  encodeAbiParameters,
+  getAbiItem,
+  hexToBytes,
+} from "viem";
+
+import { env } from "./env";
+import { Log } from "./log";
+
+require("cbor-rn-prereqs"); // Make CBOR work in React Native
+const cbor = require("cbor");
+
+// Parses authenticatorData buffer to struct
+// https://www.w3.org/TR/webauthn-2/#sctn-authenticator-data
+function parseMakeCredAuthData(buffer: Uint8Array) {
```

Comment:
i'd add a unit test for this

it's doing a lot of tricky parsing

Reply:
done